pkg/tfgen/generate_nodejs.go

```diff
@@ -551,7 +551,7 @@ func (g *nodeJSGenerator) emitResourceType(mod *module, res *resourceType) (stri
 		w.Writefmtln("        let inputs: pulumi.Inputs = {};")
 		// The lookup case:
 		w.Writefmtln("        if (opts && opts.id) {")
-		w.Writefmtln("            const state: %[1]s = argsOrState as %[1]s | undefined;", stateType)
+		w.Writefmtln("            const state = argsOrState as %[1]s | undefined;", stateType)
```

Comment:
this was failing typechecking because we effectively were doing `const state: string = blah as string | undefined`.  It's now just `const state = blah as string | undefined`.